beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1480,12 +1480,18 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
                     vec![]
                 } else {
                     vec![
+                        // TODO(hdiff): No longer relevant under HDiff hot
                         if state.slot() % T::EthSpec::slots_per_epoch() == 0 {
                             StoreOp::PutState(state_root, &state)
                         } else {
                             StoreOp::PutStateSummary(
                                 state_root,
-                                HotStateSummary::new(&state_root, &state)?,
+                                // Doing this computation here feels spagetti
+                                HotStateSummary::new(
+                                    &state_root,
+                                    &state,
+                                    &chain.store.hierarchy_hot,
+                                )?,
                             )
                         },
```

Comment:
This could be just `store.store_hot_state(..)` now, adding the temporary flag in the same batch.

Reply:
Fixed with 258406efe